wallet/udb/testdata/v9.go

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) 2017 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+// This file should compiled from the commit the file was introduced, otherwise
+// it may not compile due to API changes, or may not create the database with
+// the correct old version.  This file should not be updated for API changes.
+
+package main
+
+import (
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"encoding/binary"
+	"github.com/decred/dcrd/chaincfg"
+	"github.com/decred/dcrd/chaincfg/chainhash"
+	"github.com/decred/dcrd/dcrutil"
+	"github.com/decred/dcrd/hdkeychain"
+	"github.com/decred/dcrd/wire"
+	"github.com/decred/dcrwallet/wallet/udb"
+	"github.com/decred/dcrwallet/walletdb"
+	_ "github.com/decred/dcrwallet/walletdb/bdb"
+	"github.com/decred/dcrwallet/walletseed"
+	"golang.org/x/crypto/ripemd160"
+	"github.com/decred/dcrd/txscript"
+	"github.com/decred/dcrwallet/errors"
+	"github.com/decred/dcrd/dcrec/secp256k1"
+)
+
+const dbname = "v9.db"
+
+var (
+	epoch           time.Time
+	pubPass         = []byte("public")
+	privPass        = []byte("private")
+	privKey         = []byte{31: 1}
+	byteOrder       = binary.BigEndian
+	bucketMultisig  = []byte("ms")
+	bucketTxRecords = []byte("t")
+)
+
+func main() {
+	err := setup()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "setup: %v\n", err)
+		os.Exit(1)
+	}
+	err = compress()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "compress: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func setup() error {
+	const op errors.Op = "udb.V9DatabaseSetup"
+
+	var chainParams = &chaincfg.TestNet2Params
+	os.Remove(dbname)
+	db, err := walletdb.Create("bdb", dbname)
+	if err != nil {
+		return err
+	}
+	defer db.Close()
+	seed, err := walletseed.GenerateRandomSeed(hdkeychain.RecommendedSeedLen)
+	if err != nil {
+		return err
+	}
+	err = udb.Initialize(db, chainParams, seed, pubPass, privPass)
+	if err != nil {
+		return err
+	}
+
+	amgr, txmgr, _, err := udb.Open(db, chainParams, pubPass)
+	if err != nil {
+		return err
+	}
+
+	return walletdb.Update(db, func(dbtx walletdb.ReadWriteTx) error {
+		amgrns := dbtx.ReadWriteBucket([]byte("waddrmgr"))
+		txmgrns := dbtx.ReadWriteBucket([]byte("wtxmgr"))
+
+		err := amgr.Unlock(amgrns, privPass)
+		if err != nil {
+			return err
+		}
+
+		_, pk := secp256k1.PrivKeyFromBytes(privKey)
+
+		// Add a block
+		prevBlock := chainParams.GenesisHash
+		buf := bytes.Buffer{}
+		err = (&wire.BlockHeader{
+			Version:      1,
+			PrevBlock:    *prevBlock,
+			StakeVersion: 1,
+			VoteBits:     1,
+			Height:       uint32(1),
+		}).Serialize(&buf)
+		if err != nil {
+			return err
+		}
+
+		headerData := udb.BlockHeaderData{
+			BlockHash: chainhash.Hash{31: byte(1)},
```

Comment:
the blockhash can be created from the faked header, so use that instead of making up a block hash